packages/sdk/tests/backend_nonmock.test.ts

```diff
@@ -0,0 +1,72 @@
+import { NomadContext, NomadMessage } from '../src/index';
+import {GoldSkyBackend, GoldSkyMessage} from '../src/messageBackend/goldsky';
+
+describe('Backend internals', () => {
+    it('Backend query data only when needed', async () => {
+
+        let context = new NomadContext('production').withDefaultBackend()
+        let backend = new GoldSkyBackend('production', 'rerfg', context);
+        context._backend = backend;
+
+        const dispatchSpy = jest.spyOn(backend, 'getDispatches');
+        const getMessageSpy = jest.spyOn(backend, 'getMessage');
+
+        backend.fetchMessages = jest.fn().mockReturnValue([
+            {
```

Comment:
Can add this mocked return value to a `fixtures.ts` file. Fine for now, but if we end up re-using this since it's pretty long

Reply:
Addressed in the new commit